### PR TITLE
Enable securityContext for exporter-kube-state and operator-init

### PR DESCRIPTION
**Pull request by SeanThomasWilliams**
Update securityContext for monitoring containers. Allows monitoring to run when Rancher hardening rules are applied.

https://github.com/rancher/rancher/issues/20884

### CODE REVIEW

File: charts/rancher-monitoring/v0.0.4/charts/operator-init/templates/job-install-crds.yaml [hunk 1]
```diff
@@ -16,6 +16,10 @@ spec:
       tolerations:
 {{- include "linux-node-tolerations" . | nindent 8}}
       serviceAccountName: {{ template "app.fullname" . }}
+      securityContext:
+        runAsUser: 1000
+        runAsNonRoot: true
+        fsGroup: 2000
```
Comment (gitlawr) [inline charts/rancher-monitoring/v0.0.4/charts/operator-init/templates/job-install-crds.yaml]: UID 1000 and GID 2000 do not exist in /etc/passwd. let's use 65534(nobody). See https://github.com/helm/charts/blob/478441f4ba2ddfd55ebf3139f9c5f85fb462efa5/stable/kube-state-metrics/values.yaml#L59-L62

File: charts/rancher-monitoring/v0.0.4/values.yaml [hunk 2]
```diff
@@ -198,7 +198,10 @@ exporter-kube-state:
   ## Already exist ServiceAccount
   ##
   serviceAccountName: ""
-  securityContext: {}
+  securityContext:
+    runAsUser: 1000
+    runAsNonRoot: true
+    fsGroup: 2000
```
Comment (gitlawr) [inline charts/rancher-monitoring/v0.0.4/values.yaml]: Ditto

File: charts/rancher-monitoring/v0.0.4/charts/operator-init/templates/job-install-crds.yaml (hunk 1 above)
Comment (SeanThomasWilliams) [reply charts/rancher-monitoring/v0.0.4/charts/operator-init/templates/job-install-crds.yaml]: Thanks, updated the PR

File: charts/rancher-monitoring/v0.0.4/values.yaml (hunk 2 above)
Comment (SeanThomasWilliams) [reply charts/rancher-monitoring/v0.0.4/values.yaml]: Thanks, updated the PR

**Review by gitlawr — APPROVED** [review]
LGTM

**Comment by alena1108** [comment]
@gitlawr don't we need a new chart? v0.0.4 is already merged to release/v2.3

**Comment by gitlawr** [comment]
We need a new chart version for the changes now.
@SeanThomasWilliams Can you Update the PR and rebase these changes to the v0.0.5 monitoring chart?

**Comment by gitlawr** [comment]
Substituted by https://github.com/rancher/system-charts/pull/114

**Pull request closed without merging**